### PR TITLE
search: allow author search to perform query string query

### PR DESCRIPTION
**Pull request by MJedr**
* INSPIR-3587

### CODE REVIEW

File: backend/inspirehep/search/api.py [hunk 1]
```diff
@@ -194,15 +194,22 @@ class Meta:
...
-        names_analyzed_query = Q("match", names_analyzed=query_string)
-        names_analyzed_initials_query = Q("match", names_analyzed_initials=query_string)
-        query_string = Q("query_string", query=query_string)
-        query = Q(
-            "bool",
-            should=[names_analyzed_query, names_analyzed_initials_query, query_string],
-        )
-        return self.query(query)
+        query_string_query = Q("query_string", query=query_string)
+        if ":" in query_string:
```
Comment (drjova) [inline backend/inspirehep/search/api.py:198]: it's the opposite, if there is `:` do a normal query :)

File: backend/inspirehep/search/api.py [hunk 2]
```diff
@@ -194,15 +194,22 @@ class Meta:
...
+            query = Q(
+                "bool",
+                should=[
+                    names_analyzed_query,
+                    names_analyzed_initials_query,
+                    query_string_query,
+                ],
+            )
+            return self.query(query)
+        return self.query(query_string_query)
```
Comment (drjova) [inline backend/inspirehep/search/api.py:212]:
```suggestion
        return self.query("query_string", query=query_string)
```

and remove https://github.com/inspirehep/inspirehep/pull/1024/files#diff-d1f02c1c80adf7a3e7d52ce17ea54705R197

Comment (MJedr) [reply backend/inspirehep/search/api.py:212]: But it is also used in here: https://github.com/inspirehep/inspirehep/blob/55be643abe324762b3241c714da7e4508cc1db44/backend/inspirehep/search/api.py#L203, so either I define this query twice for each case, or once as now

Comment (drjova) [reply backend/inspirehep/search/api.py:212]: My bad I missed that!

**Pull request merged**